glean_parser/ruby_server.py

```diff
@@ -0,0 +1,138 @@
+        "ruby_server.jinja2",
+        filters=(
+            ("ping_class_name", ping_class_name),
+            ("metric_name", generate_metric_name),
+            ("metric_argument_name", generate_metric_argument_name),
+            ("metric_argument_description", generate_metric_argument_description),
+            ("event_class_name", event_class_name),
+        ),
+    )
+
+    # In this environment we don't use a concept of measurement window for collecting
+    # metrics. Only "events as pings" are supported.
+    # For each ping we generate code which contains all the logic for assembling it
+    # with metrics, serializing, and submitting. Therefore we don't generate classes for
+    # each metric as in standard outputters.
+    PING_METRIC_ERROR_MSG = (
+        " Server-side environment is simplified and only supports the events ping type"
+        + " You should not be including pings.yaml with your parser call"
+    )
+    if "pings" in objs:
+        print("❌ ping definition found." + PING_METRIC_ERROR_MSG)
+        return
+
+    # Go through all metrics in objs and build a map of
+    # ping->list of metric categories->list of metrics
+    # for easier processing in the template.
+    ping_to_metrics: Dict[str, Dict[str, List[metrics.Metric]]] = defaultdict(dict)
+    for _category_key, category_val in objs.items():
+        for _metric_name, metric in category_val.items():
+            if isinstance(metric, metrics.Metric):
+                if metric.type not in SUPPORTED_METRIC_TYPES:
+                    print(
+                        "❌ Ignoring unsupported metric type: "
+                        + f"{metric.type}:{metric.name}."
+                        + " Reach out to Glean team to add support for this"
+                        + " metric type."
+                    )
+                    continue
+                for ping in metric.send_in_pings:
+                    metrics_by_type = ping_to_metrics[ping]
+                    metrics_list = metrics_by_type.setdefault(metric.type, [])
+                    metrics_list.append(metric)
+    if "event" not in ping_to_metrics["events"]:
+        print("❌ no events found...at least one event metric is required")
+        return
+    extension = ".rb"
+    filepath = output_dir / ("server_events" + extension)
+    with filepath.open("w", encoding="utf-8") as fd:
+        fd.write(
+            template.render(
+                parser_version=__version__,
+                pings=ping_to_metrics,
+                lang=lang,
+            )
+        )
+
+
+def output_ruby(
+    objs: metrics.ObjectTree, output_dir: Path, options: Optional[Dict[str, Any]] = None
+) -> None:
+    """
+    Given a tree of objects, output ruby code to `output_dir`.
+
+    :param objects: A tree of objects (metrics and pings) as returned from
+        `parser.parse_objects`.
+    :param output_dir: Path to an output directory to write to.
+    """
+
+    output("ruby", objs, output_dir)
```

Comment:
This is a minimal wrapper around `output` and just passes on the static `"ruby"` string which is unsed.
You can remove this and rename the above function to `output_ruby` instead.

```suggestion
```

Reply:
implemented the changes you suggested and updated tests to cover